jbmc/src/java_bytecode/remove_java_new.cpp

```diff
@@ -325,24 +325,29 @@ goto_programt::targett remove_java_newt::lower_java_new(
   goto_programt &goto_program,
   goto_programt::targett target)
 {
-  const auto maybe_code_assign =
-    expr_try_dynamic_cast<code_assignt>(target->code);
-  if(!maybe_code_assign)
+  if(!target->is_assign())
     return target;
 
-  const exprt &lhs = maybe_code_assign->lhs();
-  const exprt &rhs = maybe_code_assign->rhs();
+  const auto code_assign = target->get_assign();
```

Comment:
Shouldn't this also be `const auto &`?

Reply:
of course

Reply:
Ah, no; actually doesn't work, and I now remember why.
I have added an explanation.